memory/src/main/java/org/opensearch/ml/memory/index/InteractionsIndex.java

```diff
@@ -340,4 +346,89 @@ public void deleteConversation(String conversationId, ActionListener<Boolean> li
         }
     }
 
+    /**
+     * Execute a search query over the interactions of a conversation by constructing a wrapper
+     * boolean query around the original query, AND a term query over conversation id
+     * @param conversationId the id of the conversation to query over
+     * @param request the original search request
+     * @param listener receives the search response from this query
+     */
+    public void searchInteractions(String conversationId, SearchRequest request, ActionListener<SearchResponse> listener) {
+        conversationMetaIndex.checkAccess(conversationId, ActionListener.wrap(access -> {
+            if (access) {
+                try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+                    ActionListener<SearchResponse> internalListener = ActionListener.runBefore(listener, () -> threadContext.restore());
+                    request.indices(indexName);
+                    QueryBuilder originalQuery = request.source().query();
+                    BoolQueryBuilder newQuery = new BoolQueryBuilder();
+                    newQuery.must(originalQuery);
+                    newQuery.must(new TermQueryBuilder(ConversationalIndexConstants.INTERACTIONS_CONVERSATION_ID_FIELD, conversationId));
+                    request.source().query(newQuery);
+                    client.admin().indices().refresh(Requests.refreshRequest(indexName), ActionListener.wrap(refreshResponse -> {
```

Comment:
Is 'refresh' necessary?  Does it force OpenSearch to do something every time or is it smart enough not to do anything if the index hasn't changed.